hudi-utilities/src/main/java/org/apache/hudi/utilities/deltastreamer/SourceFormatAdapter.java

```diff
@@ -50,10 +61,55 @@ public final class SourceFormatAdapter implements Closeable {
 
   private final Source source;
 
+  private Option<BaseQuarantineTableWriter> quarantineTableWriterInterface = Option.empty();
+
   public SourceFormatAdapter(Source source) {
     this.source = source;
   }
 
+  public SourceFormatAdapter(Source source, Option<BaseQuarantineTableWriter> quarantineTableWriterInterface) {
```

Comment:
can u chain the constructor. the other cnstr (L66) should call this one w/ Option.empty() as last arg

Reply:
Addressed in latest commit.